subgraph/src/public-lock.ts

```diff
@@ -50,17 +55,18 @@ function newKey(event: TransferEvent): void {
   }
 
   // update lockDayData
-  const dayID = event.block.timestamp.toI32() / 86400
-  const unlockDailyData = UnlockDailyData.load(dayID.toString())
-  if (unlockDailyData) {
-    const activeLocks = unlockDailyData.activeLocks
-    unlockDailyData.keysSold = unlockDailyData.keysSold.plus(BigInt.fromI32(1))
-    if (activeLocks && !activeLocks.includes(event.address)) {
-      activeLocks.push(event.address)
-      unlockDailyData.activeLocks = activeLocks
-    }
-    unlockDailyData.save()
+  const timestamp = event.block.timestamp.toI32()
+  const unlockDailyData = loadOrCreateUnlockDailyData(timestamp)
```

Comment:
does `event.block.timestamp.toI32()` directly gives a day? should be in seconds so there is one missing step here

Reply:
We compute the day from the timestamp in `loadOrCreateUnlockDailyData`: `  const dayID = timestamp / 86400
`